tests/model/test_model.py

```diff
@@ -1757,6 +1764,66 @@ def _set_topics_to_old_and_new(event):
             view.left_panel.show_topic_view.assert_called_once_with(stream_button)
             model.controller.update_screen.assert_called_once_with()
 
+    @pytest.mark.parametrize("original_compose_topic", ["old subject", "other subject"])
+    @pytest.mark.parametrize("original_compose_stream_id", [1, 2])
+    @pytest.mark.parametrize(
+        "propagate_mode, message_ids",
+        [("change_one", [1]), ("change_later", [1, 2]), ("change_all", [1, 2])],
+    )
+    @pytest.mark.parametrize("compose_box_open", [False, True])
+    def test_update_topic_header_mid_compose(
+        self,
+        mocker,
+        model,
+        propagate_mode,
+        message_ids,
+        compose_box_open,
+        original_compose_topic,
+        original_compose_stream_id,
+        update_message_event_index_factory,
+    ):
+        event = {
+            "type": "update_message",
+            "message_id": 1,
+            "message_ids": message_ids,
+            "orig_subject": "old subject",
+            "subject": "new subject",
+            "propagate_mode": propagate_mode,
+            "stream_id": 1,
+        }
+
+        model.index = update_message_event_index_factory(
+            msg_stream_id=1,
+            topics={1: ["old subject", "other subject"], 2: []},
+            message_ids=[1, 2],
```

Comment:
Just glancing at this alone, it's not as obvious as it could be what this translates to:
* topics -> topics_per_stream_id?
* the message_ids and msg_stream_id are coupled? If so, I'd prefer to see the data coupled and not repeated
* what are the message_ids in? If it's a default name, then we don't need to specify it, perhaps...

...or ultimately, maybe just one nested structure could work?
`{stream_id: {"old_subject": [1, 2], "other_subject": []}`

Not all of that may be necessary to get this PR bootstrapped, but it's definitely the case that we should avoid reusing numbers for streams and message ids to avoid confusion :)